wave_matching/include/wave/matching/multi_matcher.hpp

```diff
@@ -0,0 +1,83 @@
+/** @file
+ * @ingroup matching
+ *
+ * @defgroup multi_matcher
+ * Class to hold multiple instances of a matcher and multithread matches
+ */
+
+#ifndef WAVE_MULTI_MATCHER_HPP
+#define WAVE_MULTI_MATCHER_HPP
+
+#include <thread>
+#include <mutex>
+#include <condition_variable>
+#include <queue>
+#include <tuple>
+#include "wave/utils/math.hpp"
+#include "wave/matching/pcl_common.hpp"
+#include "wave/matching/matcher.hpp"
+
+namespace wave {
+/** @addtogroup matching
+ *  @{ */
+
+/**
+ * Class is templated for different matcher types
+ * @tparam T matcher type
+ * @tparam R matcher params type
+ */
+template <typename T, typename R>
+class MultiMatcher {
+ public:
+    MultiMatcher(int n_threads = std::thread::hardware_concurrency(),
+                 int queue_s = 10,
```

Comment:
Yes, but it is a default argument so it can easily be overidden. Seems like a waste to make a parameter struct for a single parameter.